addons/mass_mailing/models/mailing.py

```diff
@@ -470,47 +470,24 @@ def _get_seen_list(self):
         self.ensure_one()
         target = self.env[self.mailing_model_real]
 
-        # avoid loading a large number of records in memory
-        # + use a basic heuristic for extracting emails
         query = """
-            SELECT lower(substring(t.%(mail_field)s, '([^ ,;<@]+@[^> ,;]+)'))
+            SELECT s.email
```

Comment:
:thinking: yes that makes a lot of sense. Seems like mail_mail_statistics didn't yet hold the email when this was added at ca989b6548e93c25f54684e5569ed3998161acce which explains why it's so convoluted...

I couldn't think of a case where this would not work so fine by me.

Reply:
Note that since we fetch "email" could be worth check that we compare emails using their normalize version afterwards. You could have a trace for ``"Raoulette" <raoulette@example.fr>"`` and trying to send an email for ``"Raoulette Poiluchette" <raoulette@example.com>``. We try to correctly handle normalized emails but this is not always guaranteed, especially in 14.